pathfinder/tests/test_graphs.py

```diff
@@ -27,3 +38,69 @@ def test_routing_benchmark(token_networks: List[TokenNetwork], populate_token_ne
     print(paths)
     print(np.mean(np.array(times)), np.min(np.array(times)), np.max(np.array(times)))
     print("total_runtime = {}".format(end-start))
+
+
+def test_routing_simple(
+    token_networks: List[TokenNetwork],
+    populate_token_networks_simple: None,
+    addresses: List[Address]
+):
+    token_network = token_networks[0]
+
+    view01: ChannelView = token_network.G[addresses[0]][addresses[1]]['view']
+    view10: ChannelView = token_network.G[addresses[1]][addresses[0]]['view']
+
+    assert view01.deposit == 100
+    assert view01.transferred_amount == 20
+    assert view01.locked_amount == 0
+    assert isclose(view01.fee, 0.001)
+    assert view01.capacity == 90
+    assert view10.capacity == 60
+
+    # 0->1->4->3 is as short as 0->1->2->3 but the shortcut 1->4 is a lot more expensive.
+    # 0->2->3 would be shorter but 0->2 is degraded.
+    paths = token_network.get_paths(addresses[0], addresses[3], 10, 1, hop_bias=0)
+    assert len(paths) == 1
+    assert paths[0] == [addresses[0], addresses[1], addresses[2], addresses[3]]
+
+    # Bottleneck should be 0->1 and 2->3 with a capacity of 90.
+    with pytest.raises(NetworkXNoPath):
+        token_network.get_paths(addresses[0], addresses[3], 100, 1)
+
+    # Not connected.
+    with pytest.raises(NetworkXNoPath):
+        token_network.get_paths(addresses[0], addresses[5], 10, 1)
+
+
+def test_routing_disjoint(
+    token_networks: List[TokenNetwork],
+    populate_token_networks_simple: None,
+    addresses: List[Address],
+    monkeypatch: MonkeyPatch
+):
+    token_network = token_networks[0]
+
+    # Paths should be "as disjoint as possible". Since only two different paths exist, the first
+    # path should also be third again.
+    monkeypatch.setattr(pathfinder.token_network, 'DIVERSITY_PEN_DEFAULT', 1)
+    paths = token_network.get_paths(addresses[0], addresses[2], 10, 3)
+    assert len(paths) == 3
```

Comment:
We talked about that in the zoom. I think it would be better to not return duplicate paths.

Reply:
I will add an abort condition to `get_paths` that does just that.